src/coreclr/jit/fgehopt.cpp

```diff
@@ -3003,11 +3009,15 @@ BasicBlock* Compiler::fgCloneTryRegion(BasicBlock* tryEntry, CloneTryInfo& info,
         if (block->hasHndIndex())
         {
             const unsigned originalHndIndex = block->getHndIndex();
+            unsigned       cloneHndIndex    = originalHndIndex;
+
+            if (originalHndIndex < enclosingHndIndex)
+            {
+                cloneHndIndex += indexShift;
+            }
 
-            // if (originalHndIndex ==
-            const unsigned  cloneHndIndex = originalHndIndex + indexShift;
-            EHblkDsc* const originalEbd   = ehGetDsc(originalHndIndex);
-            EHblkDsc* const clonedEbd     = ehGetDsc(cloneHndIndex);
+            EHblkDsc* const originalEbd = ehGetDsc(originalHndIndex);
+            EHblkDsc* const clonedEbd   = ehGetDsc(cloneHndIndex);
```

Comment:
Are these `ehGetDsc` calls dead code? Ditto the `ehGetDsc` calls above for the try region logic.

Reply:
Yeah, looks like they are. 

This code has been churned quite a bit... I am still fighting off the temptation to revise the entire EH model since it is currently not update friendly. In retrospect perhaps that would have been an easier path.

Reply:
I'll fix that in a subsequent PR.